internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -543,25 +543,23 @@ func (c *enrollCmd) enrollWithBackoff(ctx context.Context, persistentConfig map[
 	defer close(signal)
 	backExp := c.backoffFactory(signal)
 
+RETRYLOOP:
 	for {
-		retry := false
 		switch {
 		case errors.Is(err, fleetapi.ErrTooManyRequests):
 			c.log.Warn("Too many requests on the remote server, will retry in a moment.")
-			retry = true
 		case errors.Is(err, fleetapi.ErrConnRefused):
 			c.log.Warn("Remote server is not ready to accept connections(Connection Refused), will retry in a moment.")
-			retry = true
 		case errors.Is(err, fleetapi.ErrTemporaryServerError):
 			c.log.Warnf("Remote server failed to handle the request(%s), will retry in a moment.", err.Error())
-			retry = true
+		case errors.Is(err, context.Canceled), errors.Is(err, context.DeadlineExceeded), err == nil:
+			break RETRYLOOP
```

Comment:
I understand that this relies on `c.enroll()` returning one of these errors if the context is cancelled.  Wouldn't it be more explicit to directly check for context cancellation here instead?

```suggestion
		case <-ctx.Done():
			break RETRYLOOP
```

Reply:
Request is done with the fleetapi, it handles context:
https://github.com/elastic/elastic-agent/blob/6b16e9cf85723c8e77ec110af0d9717b1a13dda0/internal/pkg/fleetapi/enroll_cmd.go#L210

having a `ctx.Err()` check here now means that the switch statement checks two things: `err` and `ctx.Err()`; the function eventually returns `err`